packages/smooth_app/lib/pages/home_page.dart

```diff
@@ -80,279 +80,36 @@ class _HomePageState extends State<HomePage> {
   }
 }
 
-// Note: In addition to the history of scanned products, this page currently
-// contains the user's lists and other stuff from the old Smoothie home page.
-// The intention is to move those features to other screens later or remove them
-// altogether.
-class OldHomePage extends StatefulWidget {
-  const OldHomePage();
-
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
   @override
-  State<OldHomePage> createState() => _OldHomePageState();
+  State<HistoryPage> createState() => _HistoryPageState();
 }
 
-class _OldHomePageState extends State<OldHomePage> {
-  static const ColorDestination _COLOR_DESTINATION_FOR_ICON =
-      ColorDestination.SURFACE_FOREGROUND;
-  static const Icon _ICON_ARROW_FORWARD = Icon(Icons.arrow_forward);
-
-  late DaoProductList _daoProductList;
+class _HistoryPageState extends State<HistoryPage> {
+  late DaoProductList _daoProductList ;
+  final productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+  var loadedList = false;
 
   @override
   Widget build(BuildContext context) {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
     _daoProductList = DaoProductList(localDatabase);
-    final ThemeData themeData = Theme.of(context);
-    final ColorScheme colorScheme = themeData.colorScheme;
+    if (!loadedList) {
+      loadedList = true;
+      _daoProductList.get(productList).then((_) {
```

Comment:
you may want to update the state variable `loadedList` after _daoProductList.get returns. Something like:


      _daoProductList.get(productList).then((_) {
       setState(() {
          loadedList = true;
      });
    });

Reply:
Fair enough. I had this idea that I wanted to avoid doing multiple remote fetches in case we got many renders in a row, but this feels truer - to only update state within the setState function.